src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -186,6 +186,7 @@ namespace Microsoft.Diagnostics.Tracing
 #else
 namespace System.Diagnostics.Tracing
 {
+    [Conditional("NEEDED_FOR_SOURCE_GENERATOR_ONLY")]
     [AttributeUsage(AttributeTargets.Class)]
     internal sealed class EventSourceAutoGenerateAttribute : Attribute
```

Comment:
I assume it is OK that the `EventSourceAutoGenerateAttribute` class itself will be left in the assembly? With your previous PR, the class would have been trimmed as well.

Reply:
If there is no reference to it and it's internal it should be removed by linker during build

Reply:
Ah, good point.